language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/json/JsonLfReader.java

```diff
@@ -282,81 +290,81 @@ public static <T> JsonLfDecoder<T> variant(
     // Provides a generic way to read a record type, with a constructor arg for each field.
     // This is a little fragile, so is better used by code-gen. Specifically:
     // - The constructor must cast the elements and pass them to the T's constructor appropriately.
-    // - The elements of fieldNames should all evaluate to non non-null when applied to
-    // fieldsByName.
-    // - The argIndex field values should correspond to the args passed to the constructor.
+    // - The elements of argNames should all evaluate to non non-null when applied to
+    // argsByName.
+    // - The index field values should correspond to the args passed to the constructor.
     //
     // e.g.
     //     r.record(
     //        asList("i", "b"),
     //        name -> {
     //          switch (name) {
     //            case "i":
-    //              return JsonLfReader.Field.at(0, r.list(r.int64()));
+    //              return JsonLfReader.ConstrArg.at(0, r.list(r.int64()));
```

Comment:
the use of constructor trips me up a bit. In LF we usually only talk about constructors for variants and enums. don't feel strongly about it though so fine to keep it.

Reply:
Renamed to `JavaArg`.